ast/compile_test.go

```diff
@@ -875,6 +875,139 @@ func TestCompilerCheckTypesWithSchema(t *testing.T) {
+			schema:        allOfObjectMissing,
+			expectedError: nil,
+		},
+		{
+			note:          "allOf with Array of mergeable different types in schema",
+			schema:        allOfArrayDifTypes,
+			expectedError: nil,
+		},
+		{
+			note:          "allOf with mergeable Object containing Array types in schema",
+			schema:        allOfArrayInsideObject,
+			expectedError: nil,
+		},
+		{
+			note:          "allOf with mergeable Array types in schema with type declaration missing",
+			schema:        allOfArrayMissing,
+			expectedError: nil,
+		},
+		{
+			note:          "allOf with mergeable types inside of core schema",
+			schema:        allOfInsideCoreSchema,
+			expectedError: nil,
+		},
+		{
+			note:          "allOf with mergeable String types in schema",
+			schema:        allOfStringSchema,
+			expectedError: nil,
+		},
+		{
+			note:          "allOf with mergeable Integer types in schema",
+			schema:        allOfIntegerSchema,
+			expectedError: nil,
+		},
+		{
+			note:          "allOf with mergeable Boolean types in schema",
+			schema:        allOfBooleanSchema,
+			expectedError: nil,
+		},
+		{
+			note:          "schema of array type with uneven numbers of items to merge under allOf",
+			schema:        allOfSchemaWithUnevenArray,
+			expectedError: nil,
+		},
+		{
+			note:          "allOf schema with array of arrays",
+			schema:        allOfArrayOfArrays,
+			expectedError: fmt.Errorf("unable to merge these schemas"),
+		},
+		{
+			note:          "allOf schema with array and object as siblings",
+			schema:        allOfObjectAndArray,
+			expectedError: fmt.Errorf("unable to merge these schemas"),
+		},
+		{
+			note:          "allOf array schema with different types within arrays to be merged",
+			schema:        allOfArrayDifTypesWithError,
+			expectedError: fmt.Errorf("unable to merge these schemas"),
+		},
+		{
+			note:          "allOf schema with different types",
+			schema:        allOfTypeErrorSchema,
+			expectedError: fmt.Errorf("unable to merge these schemas"),
+		},
+		{
+			note:          "allOf unmergeable schema with different type parent and children",
+			schema:        allOfSchemaWithParentError,
+			expectedError: fmt.Errorf("unable to merge these schemas"),
+		},
+		{
+			note:          "allOf schema of array type with uneven numbers of items to merge",
+			schema:        allOfSchemaWithUnevenArray,
+			expectedError: nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.note, func(t *testing.T) {
+			c := NewCompiler()
+			var schema interface{}
+			err := util.Unmarshal([]byte(tc.schema), &schema)
+			if err != nil {
+				t.Fatal("Unexpected error:", err)
+			}
+			schemaSet := NewSchemaSet()
+			schemaSet.Put(SchemaRootRef, schema)
+			c.WithSchemas(schemaSet)
+			compileStages(c, c.checkTypes)
+			if tc.expectedError != nil {
+				errors := c.Errors
+				if !strings.Contains(errors.Error(), tc.expectedError.Error()) {
```

Comment:
You could change this to `if !strings.Contains(c.Errors.Error(), tc.expectedError.Error()) {` and remove the previous line `errors := c.Errors`